packages/eslint-plugin/src/rules/no-unnecessary-parameter-property-assignment.ts

```diff
@@ -0,0 +1,70 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+
+export const messageId = 'noUnnecessaryParameterPropertyAssignment';
+
+export default util.createRule({
+  name: 'no-unnecessary-parameter-property-assignment',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Disallow unnecessary assignment of constructor property parameter to itself',
+      recommended: 'error',
+      requiresTypeChecking: false,
+    },
+    schema: [],
+    messages: {
+      [messageId]:
```

Comment:
Maybe ... `unnecessaryAssignment`?

```suggestion
      unnecessaryAssignment:
```